samples/dubbogo/multi/server/app/student.go

```diff
@@ -227,32 +227,32 @@ func (s *StudentProvider) GetStudentByNameAndAge(ctx context.Context, name strin
 func (s *StudentProvider) UpdateStudent(ctx context.Context, Student *Student) (bool, error) {
 	println("Req UpdateStudent data:%#v", Student)
 	r, ok := studentCache.GetByName(Student.Name)
-	if ok {
-		if Student.ID != "" {
-			r.ID = Student.ID
-		}
-		if Student.Age >= 0 {
-			r.Age = Student.Age
-		}
-		return true, nil
+	if !ok {
+		return false, errors.New("not found")
+	}
+	if Student.ID != "" {
```

Comment:
when u wanna check a string or an array is nil, pls using len(Strudent.ID) instead.